osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ClearMBusStatusOnAllChannelsCommandExecutor.java

```diff
@@ -75,11 +81,30 @@ public AccessResultCode execute(
       throws ProtocolAdapterException {
 
     try {
-      for (final int channel : CHANNELS) {
-        this.clearStatusMaskForChannel(conn, channel, device);
+      final ScanMbusChannelsResponseDto scanMbusChannelsResponseDto =
+          this.adhocService.scanMbusChannels(conn, device, messageMetadata);
+
+      final List<String> exceptions = new ArrayList<>();
+
+      scanMbusChannelsResponseDto.getChannelShortIds().stream()
+          .filter(dto -> !"00000000".equals(dto.getShortId().getIdentificationNumber()))
```

Comment:
I would extract this in a _Predicate_ so you can give it all logical name explaining the check